### PR TITLE
Feature: Add breadcrumbs to association pages

### LINKED ISSUE #3062: Show breadcrumbs when viewing a `has_many` association alone
Related: https://github.com/avo-hq/avo/issues/3061

When viewing an [association page separate](https://avodemo.com/avo/resources/users/eric-berry/teams?view=show&turbo_frame=has_and_belongs_to_many_field_show_teams) you may want to have the breadcrumbs available so you can go back from the UI.

To show the breadcrumbs:
```ruby
display_breadcrumbs: @reflection.blank? || (@reflection.present? && !helpers.turbo_frame_request?)
```

Still to do. make the breadcrumbs show the parent record path.

### PR DESCRIPTION
**Pull request by binarygit**
When users open "attachment" pages, like the model for attaching a team member, as standalone pages we need to display breadcrumbs. This PR adds those breadcrumbs to such pages.

# Description

Fixes #3062

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [x] I have added tests that prove my fix is effective or that my feature works

## Screenshots & recording

When navigating to the team_members page from Teams:
image

When navigating to the teams page from Users:
image

## Manual review steps

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

**Review by Paul-Bob — CHANGES_REQUESTED** [review]
Association tables are broken when loading [a team page](http://localhost:3030/admin/resources/teams/3)

Also lets ensure tests are passing

**Review by adrianthedev — CHANGES_REQUESTED** [review]
Let's undo the `@turbo_frame` and remove `html_req_for_association_page?`

File: app/controllers/avo/base_controller.rb [hunk 1]
```diff
@@ -18,6 +18,11 @@ class BaseController < ApplicationController
 
     def index
       @page_title = @resource.plural_name.humanize
+
+      if @reflection.present? && !helpers.turbo_frame_request?
```
Comment (adrianthedev) [inline app/controllers/avo/base_controller.rb:22]: do we need `helpers.` here?

Comment (binarygit) [reply app/controllers/avo/base_controller.rb:22]: Ah! sorry I missed it.

File: app/components/avo/views/resource_index_component.html.erb [hunk 2]
```diff
@@ -11,7 +11,7 @@
       description: description,
       cover_photo: resource.cover_photo,
       data: {component: "resources-index"},
-      display_breadcrumbs: @reflection.blank?
+      display_breadcrumbs: @reflection.present? && !helpers.turbo_frame_request?
```
Comment (Paul-Bob) [inline app/components/avo/views/resource_index_component.html.erb:14]:
```suggestion
      display_breadcrumbs: @reflection.blank? || (@reflection.present? && !helpers.turbo_frame_request?)
```

**Review by Paul-Bob — CHANGES_REQUESTED** [review]

File: spec/features/avo/breadcrumbs_spec.rb [hunk 3]
```diff
@@ -21,4 +21,37 @@
     it { is_expected.to have_css ".breadcrumbs" }
     it { is_expected.to have_text "Home\n" }
   end
+
+  describe "on a has_and_belongs_to_many turbo frame" do
+    let!(:user) { create(:user) }
+    let!(:team) { create(:team) }
+
+    it "displays breadcrumbs" do
+      url = "/admin/resources/users/#{user.slug}/teams?view=show&turbo_frame=has_and_belongs_to_many_field_show_teams"
```
Comment (adrianthedev) [inline spec/features/avo/breadcrumbs_spec.rb:30]:
```suggestion
      url = "/admin/resources/users/#{user.slug}/teams?view=show"
```

Comment (binarygit) [reply spec/features/avo/breadcrumbs_spec.rb]: Done

**Pull request merged**